components/camel-jms/src/main/java/org/apache/camel/component/jms/JmsConfiguration.java

```diff
@@ -649,6 +652,7 @@ public JmsOperations createInOutTemplate(JmsEndpoint endpoint, boolean pubSubDom
                 // default to AUTO
                 jmsTemplate.setSessionAcknowledgeMode(Session.AUTO_ACKNOWLEDGE);
             }
+            jmsTemplate.setDeliveryDelay(deliveryDelay);
```

Comment:
It does not make sense to set it here again for InOut. This method calls `createInOnlyTemplate` as first statement which returns JmsTemplate with already set delay. This can be removed.